### PR TITLE
lib: nrf_modem_lib: enable library when compiling for nrf9160

### PR DESCRIPTION
**Pull request by lemrey**
In order to achieve low power consumption, the Modem
must be turned on to perform some hardware configuration steps.

Enable the Modem library, thus initializing the Modem,
in every sample that is built for the ~~nrf9160dk_nrf9160ns board~~ nRF9160 SoC .

### CODE REVIEW

File: lib/nrf_modem_lib/Kconfig [hunk 1]
```diff
@@ -9,6 +9,7 @@ menuconfig NRF_MODEM_LIB
 	imply NET_SOCKETS_OFFLOAD
 	imply NET_SOCKETS_POSIX_NAMES if !POSIX_API
 	select NRF_MODEM
+	default y if BOARD_NRF9160DK_NRF9160_NS
```
Comment (rlubos) [inline lib/nrf_modem_lib/Kconfig:12]: What about Thingy91? Or any other nRF91-based board, shall we bother? Maybe it'd be better to rely on SoC type instead? Something like:
`default y if SOC_NRF9160_SICA && TRUSTED_EXECUTION_NONSECURE`

**Review by shantha-14 — APPROVED** [review]

**Review by jtguggedal — CHANGES_REQUESTED** [review]
I think this needs a changelog entry. I think it's a pretty significant change in that i adds a pretty big library and partitions to all builds

File: lib/nrf_modem_lib/Kconfig (hunk 1 above)
Comment (lemrey) [reply lib/nrf_modem_lib/Kconfig:12]: Yeah, good point.

**Comment by greg-fer** [comment]
@jtguggedal , @lemrey , there are no changes in the modem library docs needed due to this change?

**Review by rlubos — APPROVED** [review]

**Comment by lemrey** [comment]
@jtguggedal yes, definitely. I have added a line @greg-fer please have a look. We won't need changes in the modem library documentation for this, no. The change makes it so that all samples compiled for the nrf9160 will have the modem library enabled.

**Review by jtguggedal — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Comment by lemrey** [comment]
Not sure what's up with the `crypto/psa_tls` sample, it fails with the same error that should have been fixed in the last commit I pushed. Also I can't build that sample locally:
```
$ cmake -version
cmake version 3.21.0

-- The C compiler identification is GNU 10.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/emdi/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Found Python3: /usr/local/bin/python3.8 (found version "3.8.2") found components: Interpreter 
Traceback (most recent call last):
  File "/home/emdi/git/nrfconnect/modules/tee/tfm/trusted-firmware-m/tools/tfm_parse_manifest_list.py", line 12, in <module>
    from jinja2 import Environment, BaseLoader, select_autoescape, TemplateNotFound
ModuleNotFoundError: No module named 'jinja2'
CMake Error at tools/CMakeLists.txt:95 (message):
  File generation failed

$ pip3 install --user jinja2
Requirement already satisfied: jinja2 in /usr/lib/python3/dist-packages (2.10)
```

It complains about missing `jinja2` package, which I have installed.

**Review by jhn-nordic — APPROVED** [review]

**Comment by jhn-nordic** [comment]
Tested with PPK. event manager sample in nrf/samples/ went from 3.6mA to 430uA. Looks to work as intended

**Comment by lemrey** [comment]
Manually disabling `libmodem` in a the `gps_module` test for `asset_tracker_v2`, to avoid bringing in a definition of `k_malloc` which the test re-defines.

**Review by simensrostad — APPROVED** [review]

**Pull request merged**